overlord/state/notices.go

```diff
@@ -211,6 +296,30 @@ type AddNoticeOptions struct {
 
 	// Time, if set, overrides time.Now() as the notice occurrence time.
 	Time time.Time
+
+	// RepeatCheckData is saved for stateful tracking throughout the notice's lifetime.
+	//
+	// It is intended to be accessed through RepeatCheck callback through
+	// oldNotice.GetRepeatCheckValue().
+	//
+	// NOTES:
+	//	- RepeatCheckData must be JSON marshallable.
+	//	- Setting RepeatCheckData to nil doesn't remove old state.
+	RepeatCheckData interface{}
+
+	// RepeatCheck, if set, overrides options dynamically based on the current state
+	// through updating newNoticeOpts.
+	//
+	// NOTE: Current state is intended to be accessed through oldNotice.GetRepeatCheckValue().
+	RepeatCheck func(oldNotice *Notice, newNoticeOpts *AddNoticeOptions) error
```

Comment:
I still think this should return a newRepeatCheckData and a bool, the intention was not to let this update the options